rbac/default-role-manager/role_manager_test.go

```diff
@@ -364,29 +364,47 @@ func TestTemporaryRoles(t *testing.T) {
 }
 
 func TestMaxHierarchyLevel(t *testing.T) {
-	rm := NewRoleManager(1)
```

Comment:
To maintain compatibility, I think it is best not to restrict AddLink by level, and only filter by level when gettingRoles/Uses and HasLink.